### ISSUE TITLE
vague traceback in _sslverify.py

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3692]
|<img alt="imgrey's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| imgrey reported|
|-|-|
|Trac ID|trac#3692|
|Type|enhancement|
|Created|2009-03-18 17:49:23Z|

```
Traceback (most recent call last):
  File "./create_CSR-n-sign.py", line 17, in <module>
    cert = ca_cert.newCertificate(cl_cert.dump())
  File "/usr/lib/python2.5/site-packages/Twisted-8.2.0_r26382-py2.5-linux-i686.egg/twisted/internet/_sslverify.py", line 346, in newCertificate
    return self.load(newCertData, self.privateKey, format)
  File "/usr/lib/python2.5/site-packages/Twisted-8.2.0_r26382-py2.5-linux-i686.egg/twisted/internet/_sslverify.py", line 350, in load
    return Class._load(data, format)._setPrivateKey(privateKey)
  File "/usr/lib/python2.5/site-packages/Twisted-8.2.0_r26382-py2.5-linux-i686.egg/twisted/internet/_sslverify.py", line 336, in _setPrivateKey
    "Sanity check failed: Your certificate was not properly signed.")
twisted.internet.error.VerifyError: Sanity check failed: Your certificate was not properly signed.
```

As it turned out I was mixing up certificates:
```
from twisted.internet.ssl import KeyPair, PrivateCertificate, DistinguishedName
cl_cert = PrivateCertificate.loadPEM(file('ss.pem', 'r').read())
csr = cl_cert.certificateRequest()
ca_cert = PrivateCertificate.loadPEM(file('tree.pem', 'r').read())
cert_data = ca_cert.signCertificateRequest(csr,
    lambda dn: True, 1)
cert = cl_cert.newCertificate(cert_data)
# here
# cert = ca_cert.newCertificate(cert_data)
print cert.dumpPEM()
```

Attachments:

* [better-exception.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/bb3/bb3e29ac93b725f6eefbfcf1bf0bee67783cd519/cc9c1f281872cf576f22037bcb3ffbef126c81c3.patch) (574 bytes) - added by exarkun on 2009-04-11 18:20:53Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__3692 3692
type__enhancement enhancement
reporter__imgrey imgrey
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1237398563000000 1237398563000000
changetime__1239645218000000 1239645218000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

I propose that the wording in the attached patch more clearly indicates what error has happened in this case.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve set owner to @exarkun|
|-|-|

Fair enough, +1.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [26690]) Apply better-exception.patch

Author: exarkun
Reviewer: therve
Fixes: [#3692](https://github.com/twisted/twisted/issues/3692)

Change the text of the `VerifyError` raised when an attempt is made to associate
a private key with a `PrivateCertificate` which does not match that certificate's
public key.  The text now states explicitly what the mismatch was, rather than
just making a general statement about an improperly signed certificate.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
